src/VirtoCommerce.Platform.Web/Middleware/Hangfire/HangfireUserContextMiddleware.cs

```diff
@@ -0,0 +1,81 @@
+using System;
+using System.Linq;
+using System.Threading;
+using Hangfire.Client;
+using Hangfire.Server;
+using Microsoft.AspNetCore.Http;
+using VirtoCommerce.Platform.Core.ChangeLog;
+using VirtoCommerce.Platform.Security;
+
+namespace VirtoCommerce.Platform.Web.Middleware.Hangfire
+{
+    public class HangfireUserContextMiddleware : IClientFilter, IServerFilter
+    {
+        private readonly IHttpContextAccessor _contextAccessor;
+
+        /// <summary>
+        /// Ctor
+        /// </summary>
+        public HangfireUserContextMiddleware(IHttpContextAccessor contextAccessor) => _contextAccessor = contextAccessor;
+
+        /// <summary>
+        /// Check if argument contains information about log operation
+        /// </summary>
+        private readonly Func<object, bool> AuditableArgument = arg => arg is OperationLog || arg is OperationLog[];
```

Comment:
I think it's better to fill username user for all Hangfire tasks, not just for those that are working with OperationLog.
As we have many AuditableEntities that do not have the handler which writes to operation log.
And even for those entities I think it is better to have proper user name in CreatedBy/ModifiedBy